packages/tupaia-web/src/features/Map/utils/useMapOverlayTableData.ts

```diff
@@ -86,7 +86,7 @@ export const useMapOverlayTableData = ({
 
   return {
     ...data,
-    isLoading: isLoading || isFetching || !isFetched,
+    isLoading: isLoading || isFetching || (!isFetched && !isIdle),
```

Comment:
I find these loading states are a bit frustrating. I would just expec isLoading to include all of these cases (I think it might in the latest version). I wonder if we should try and extend and modify the default loading logic somehow to do this. Maybe not a task for now though.